local_planner: field_local_planner::RmpPlugin
robot_length: 1.0
robot_width: 0.5
robot_height: 0.5
base_frame: base
fixed_frame: odom
distance_to_goal_thr: 0.1 # meters
orientation_to_goal_thr: 0.1  # radians
max_linear_velocity_x: 1.0    # m/s
max_linear_velocity_y: 1.0    # m/s
max_angular_velocity_z: 1.0   # rad/s
control_rate: 10 # Hz

rmp:
  # Control points define the points where the accelerations are exerted on the body
  control_points:
    mc :
      point_factor: [0.0, 0.0]
      affected_by_goal: true
      affected_by_obstacle: true
      color: [0.00, 0.00, 0.00]
      radius: 0.35
    flc :
      point_factor: [0.6, 0.3]
      affected_by_goal: false
      affected_by_obstacle: true
      color: [0.00, 0.00, 0.00]
      radius: 0.3
    frc :
      point_factor: [0.6, -0.3]
      affected_by_goal: false
      affected_by_obstacle: true
      color: [0.00, 0.00, 0.00]
      radius: 0.3
    blc :
      point_factor: [-0.6, 0.3]
      affected_by_goal: false
      affected_by_obstacle: true
      color: [0.00, 0.00, 0.00]
      radius: 0.3
    brc :
      point_factor: [-0.6, -0.3]
      affected_by_goal: false
      affected_by_obstacle: true
      color: [0.00, 0.00, 0.00]
      radius: 0.3
  # Tune the sphere radius used for the control points
  sphere_radius_factor: 1.2

  # Geodesic Goal RMP parameters
  # This weights the contribution of the acceleration
  geodesic_goal_weight: 1.0
  # This gain scales the force that points toward the goal
  geodesic_goal_gain: 20.0
  # This factor determines when the rmp is disabled
  geodesic_goal_offset: 0.0
  # Steepness to disable heading to goal
  geodesic_goal_steepness: 20.0
  # # Enable when close to the goal
  # rmp/goal_enable_when_close: false

  # Geodesic Flow Heading RMP parameters
  # This weights the contribution of the acceleration
  geodesic_heading_weight: 1.0
  # This gain scales the force that points toward the goal
  geodesic_heading_gain: 20.0
  # This factor determines when the rmp is disabled
  geodesic_heading_offset: 0.5 # This weights the contribution of the acceleration
  # Steepness to disable heading to goal
  geodesic_heading_steepness: 20.0
  # # Enable when close to the goal
  # rmp/goal_enable_when_close: false

  # Analytical Goal position
  # This weights the contribution of the acceleration
  goal_position_weight: 0.0
  # # This gain scales the force that corrects for the goal pose
  goal_position_gain: 20.0
  # # This factor scales the velocity to damp the force
  goal_position_offset: 0.5
  # # Steepness to enable heading to goal
  goal_position_steepness: 20.0
  # # Enable when close to the goal
  # rmp/goal_position_enable_when_close: true

  # Analytical Goal heading
  # This weights the contribution of the acceleration
  goal_heading_weight: 1.0
  # # This gain scales the force that corrects for the goal pose
  goal_heading_gain: 20.0
  # # This factor scales the velocity to damp the force
  goal_heading_offset: 0.5
  # # Steepness to enable heading to goal
  goal_heading_steepness: 20.0
  # # Enable when close to the goal
  # rmp/goal_heading_enable_when_close: true

  # Heading RMP
  # This weights the contribution of the acceleration
  heading_weight: 0.0
  # Tries to keep the robot heading in the same direction 
  # of the twist command
  # This gain scales the force changes the heading
  heading_gain: 10.0
  # This factor determines when the rmp is disabled
  heading_offset: 1.0
  # Steepness to disable heading to goal
  heading_steepness: 20.0

  # Damping RMP
  # This weights the contribution of the acceleration
  damping_weight: 1.0
  # Opposes the force by applying a damping proportional to velocity
  damping: 0.5

  # Obstacle RMP parameters
  # The obstacle RMPs are created for each potential 
  # collision point 

  # This weights the contribution of the acceleration
  obstacle_weight: 0.0
  # This gain scales the force exerted by the obstacles
  obstacle_gain: 1.0
  # Clearance distance from the obstacles
  obstacle_offset: 0.3 # meters
  # Clearance distance from the obstacles
  obstacle_steepness: 10.0 # meters

  # Regularization RMP
  # As with factor graphs, we can add a prior to smooth
  # the solution a bit. This factor scales the importance
  # of the lastly optimized acceleration which is added
  # as a prior
  regularization: 1e-6

  # RMP computes an optimal acceleration (force) that
  # Must be integrated to get a velocity
  # This basically scales the acceleration
  integration_time: 0.1
